src/generators/clientContextFileGenerator.ts

```diff
@@ -20,7 +20,8 @@ import { formatJsDocParam } from "./utils/parameterUtils";
 export function generateClientContext(
   clientDetails: ClientDetails,
   packageDetails: PackageDetails,
-  project: Project
+  project: Project,
+  hasCustomLayer: boolean
```

Comment:
Perhaps the name is a bit unclear. One alternative could be `exportedByPackage`. Either way, please document it.

Reply:
The ```hasCustomLayer``` could be used for multiple purposes in the future. Not just hiding classes. Added some documentation for the property.